src/soundio/sounddeviceportaudio.cpp

```diff
@@ -247,23 +249,37 @@ SoundDeviceError SoundDevicePortAudio::open(bool isClkRefDevice, int syncBuffers
     m_syncBuffers = syncBuffers;
 
     // Create the callback function pointer.
-    PaStreamCallback* callback = nullptr;
+    PaStreamCallback* pCallback = nullptr;
     if (isClkRefDevice) {
-        callback = paV19CallbackClkRef;
+        pCallback = paV19CallbackClkRef;
+    } else if (framesPerBuffer == paFramesPerBufferUnspecified) {
+        m_syncBuffers = 1;
+        // This happens in case of JACK, where PortAudio creates artificial
+        // device streams from one native JACK server callback "JackCallback()"
+        // For every sound hardware. Clock drift can not happen, but the buffers
+        // are processed: In 1 - Out 1 - In 2 - Out 2
+        // This causes even with the "Experimental (no delay)" setting
+        // one extra buffer delay for the non clock reference device
```

Comment:
There's no way to avoid this extra latency when using PortAudio with multiple JACK clients? If so, I think that's a good justification to not use PortAudio for JACK.

Reply:
I see no way. And yes that is only one reason why Mixxx should become a native JACK client.